ingestion_server/ingestion_server/indexer.py

```diff
@@ -312,6 +316,38 @@ def reindex(
             database_connect(), model_name, table_name, destination_index, self.task_id
         )
 
+    def create_filtered_index(self, model_name: str, index_suffix: str, **_):
+        """
+        Create a secondary index that excludes documents filtered by a list of terms.
+        """
+
+        source_index = f"{model_name}-{index_suffix}"
+        destination_index = f"{source_index}-filtered"
+
+        log.info(f"Creating filtered index '{destination_index}' from '{source_index}'")
+        self.es.reindex(
+            body={
+                "source": {
+                    "index": source_index,
+                    "query": {
+                        "bool": {
+                            "must_not": [
+                                {
+                                    "multi_match": {
+                                        "query": term,
+                                        "fields": ["tags.name", "title", "description"],
+                                    }
+                                }
+                                for term in SENSITIVE_TERMS
+                            ]
+                        }
+                    },
+                },
+                "dest": {"index": destination_index},
+            }
+        )
+        self.ping_callback()
```

Comment:
This will send the callback immediately, whereas ES will continue to process the `reindex` call on its end and create the index. This is most likely what is breaking the tests because the index is not ready by the time the test starts asking for it.